mltsp/util.py

```diff
@@ -139,14 +142,18 @@ def cast_model_params(model_type, model_params):
             if dest_type is not str:
                 try:
                     if isinstance(ast.literal_eval(model_params[k]),
-                                  dest_type):
+                                  dest_type) or \
+                        (params_to_optimize and k in params_to_optimize and \
```

Comment:
Separate logic for `params_to_optimize` once it's a `dict` instead of list of names...